### PR TITLE
Add tooltips to pipeline run nodes

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
## Fixes 
Fixes [HAC-3257](https://issues.redhat.com/browse/HAC-3257)

## Description
Add tooltips to tasks in the Pipeline run visualization. Refactors some of the data fields to be more type check-able and remove duplicate data.

## Type of change
- [x] Feature
- [x] Refactoring (no functional changes, no api changes)

## Screen shots / Gifs for design review 

image

image

/cc @karthikjeeyar @christianvogt

### CODE REVIEW

**Comment by karthikjeeyar** [comment]
@MariaLeonova I remember that Matt had a suggestion to not have the task description in the tooltip because of real estate and not as useful as status, duration etc, can you please confirm if we need this?

**Comment by karthikjeeyar** [comment]
Tested it locally, here are few observations

 
1. The substasks of a `Cancelled` task contains red (`failed`) color, should we show the same color as its parent?

Screenshot 2023-03-08 at 3 51 01 PM

2. In-progress/running icon is missing

image

**Comment by MariaLeonova** [comment]
@karthikjeeyar Correct! However, until we have a side panel, the description has to go somewhere. So for now let's have it in the tooltip, and move it once we have an opportunity to do so.

**Comment by jeff-phillips-18** [comment]
Fixed the icons. Updated how we determine the status of the task steps based on the task step's terminated.reason field.

File: config/dev.webpack.config.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ const webpackProxy = {
     {
       context: (path) => path.includes('/api/k8s/registration'),
       target:
-        'https://registration-service-toolchain-host-operator.apps.appstudio-stage.x99m.p1.openshiftapps.com',
+        'https://registration-service-toolchain-host-operator.apps.hac-dev.clusters.stonesoupengineering.com',
```
Comment (karthikjeeyar) [inline config/dev.webpack.config.js:33]: revert this changes

File: src/components/PipelineRunDetailsView/visualization/utils/pipelinerun-graph-utils.ts [hunk 2]
```diff
@@ -75,6 +121,37 @@ export const getPipelineFromPipelineRun = (pipelineRun: PipelineRunKind): Pipeli
...
+  if (!status || !status.reason) {
+    stepRunStatus = RunStatus.Cancelled;
+  } else {
+    if (!matchingStep) {
+      stepRunStatus = RunStatus.Pending;
+    } else if (matchingStep.terminated) {
+      stepRunStatus = terminatedReasonToRunStatus(matchingStep.terminated.reason);
+      duration = getStepDuration(matchingStep) || status.duration;
+    } else if (matchingStep.running) {
+      stepRunStatus = RunStatus['In Progress'];
```
Comment (karthikjeeyar) [inline src/components/PipelineRunDetailsView/visualization/utils/pipelinerun-graph-utils.ts:141]: The RunStatus that comes from pipelines topology don't have a gap in-between, either remove the gap or change it to `RunStatus.Running`

**Comment by jeff-phillips-18** [comment]
Remove the description from the tooltips.
